app/services/sso/openid_connect.rb

```diff
@@ -42,9 +42,13 @@ def bearer_token_set?
     end
 
     def valid_issuer?
+      logger = Foreman::Logging.logger('app')
       payload = jwt_token.decoded_payload
+      if payload.nil?
+        logger.error "Nil JWT received, please check connectivity with the OpenID Provider"
```

Comment:
Since `nil` payload might indicate invalid payload as well, perhaps change this to:
```suggestion
        logger.error "Invalid JWT received, please check connectivity with the OpenID Provider"
```